src/parser/jobs/mch/Heat.js

```diff
@@ -0,0 +1,271 @@
+				if (abilityId !== ACTIONS.SHOT.id) {
+					// We don't really need auto attacks in here, they'll just clog up the display
+					this._overheatWindows.current.casts.push(event)
+				}
+			} else {
+				this._overheatWindows.current.gcdCount = this._overheatWindows.current.casts.filter(cast => getAction(cast.ability.guid).onGcd).length
+				this._overheatWindows.history.push(this._overheatWindows.current)
+				this._heat = 0
+				this._setBarrelState(BARREL_STATE.COOLING, OVERHEAT_DURATION_MILLIS)
+			}
+			break
+		case BARREL_STATE.COOLING:
+			if (event.timestamp - this._lastBarrelStateChange >= COOLING_DURATION_MILLIS - COOLING_FUZZ_MILLIS) {
+				this._setBarrelState(BARREL_STATE.NORMAL, COOLING_DURATION_MILLIS)
+				if (abilityId === ACTIONS.GAUSS_BARREL.id) {
+					this._setBarrelState(BARREL_STATE.GAUSS)
+				}
+			} else if (abilityId === ACTIONS.GAUSS_BARREL.id) {
+				this._fixNullAssumption()
+			}
+			break
+		}
+	}
+
+	_onApplyFlamethrower(event) {
+		this._flamethrowerApplied = event.timestamp
+	}
+
+	_onRemoveFlamethrower(event) {
+		if (this._flamethrowerApplied > -1) {
+			// The first tick is always within a few milliseconds of the buff application and all subsequent ticks fire 1 second after the last
+			const ticks = Math.floor((event.timestamp - this._flamethrowerApplied) / FLAMETHROWER_TICK_MILLIS) + 1
+			this._addHeat(ticks * FLAMETHROWER_TICK_HEAT_GAIN)
+			this._flamethrowerApplied = -1
+		}
+	}
+
+	_onDeath() {
+		this._heat = 0
+		this._setBarrelState(BARREL_STATE.NORMAL)
+	}
+
+	_onComplete() {
+		this._setBarrelState(BARREL_STATE.NORMAL) // So we tally the barrel state for the last chunk of the fight
+		const averageGaussDowntime = this._barrelStateTime[BARREL_STATE.NORMAL] / this._overheatWindows.history.length
+		this.suggestions.add(new TieredSuggestion({
+			icon: ACTIONS.GAUSS_BARREL.icon,
+			content: <Fragment>
+				<Trans id="mch.heat.suggestions.downtime.content">Make sure to use <ActionLink {...ACTIONS.GAUSS_BARREL}/> as soon as it's available after overheat/cooldown windows, as you can't build up heat without it.</Trans>
+			</Fragment>,
+			tiers: {
+				1000: SEVERITY.MINOR,
+				3000: SEVERITY.MEDIUM,
+			},
+			value: averageGaussDowntime,
+			why: <Fragment>
+				<Trans id="mch.heat.suggestions.downtime.why">Gauss Barrel was down for {this.parser.formatDuration(this._barrelStateTime[BARREL_STATE.NORMAL])}.</Trans>
+			</Fragment>,
+		}))
+
+		this.suggestions.add(new TieredSuggestion({
+			icon: ACTIONS.COOLDOWN.icon,
+			content: <Fragment>
+				<Trans id="mch.heat.suggestions.cooldown.content">Avoid using <ActionLink {...ACTIONS.COOLDOWN}/> if it would bring you below {HEATED_POINT} heat. Dipping too low will cost you heated shots and may delay overheat windows, throwing off your rotation and OGCD alignment.</Trans>
+			</Fragment>,
+			tiers: {
+				1: SEVERITY.MEDIUM,
+				2: SEVERITY.MAJOR,
+			},
+			value: this._badCooldowns,
+			why: <Fragment>
```

Comment:
You can scrap the `<Fragment>` in this case.

Reply:
Honestly I should probably do that on most of my modules. I think I have hella unnecessary `<Fragment>` tags all over the place right now.